app/app/api/api_v1/endpoints/students.py

```diff
@@ -61,6 +71,30 @@ def read_student_by_id(
     raise ForbiddenException(detail="The user doesn't have enough privileges")
 
 
+@router.get("/{student_id}/divisions", response_model=List[schemas.Division])
+def read_student_divisions_by_id(
+    student_id: str, current_user: models.User = Depends(deps.get_current_user), db: Session = Depends(deps.get_db)
+) -> Any:
+    """
+    Get a specific student's divisions by ID.
+    """
+    # Fetch student with the corresponding ID from DB
+    if student := crud.student.get(db, id=student_id):
+        # Return the fetched divisions if current_user is trying to fetch itself
+        # or is an admin with the required perms
+        if current_user.id == student_id or (
+            (admin := crud.admin.get(db, id=current_user.id))
+            and AdminPermissions(admin.permissions).is_allowed("student")
+        ):
+            return list(student.divisions)
+
+        raise ForbiddenException(detail="The user doesn't have enough privileges")
+
+    raise NotFoundException(
```

Comment:
Hmm should this also have a permissions check, just so you someone without permissions can't spam random IDs and find out whether they exist or not.

Reply:
I mean, with UUIDs, what are the chances of randomly getting a success? Enough to go and change all API endpoints where someone can check if a certain object exists?

Reply:
Ideally, yes😂
But realistically, just wanted to discuss, I think it is done in some modules already. Not a priority really, can merge as it is.